### PR TITLE
Feature/cq v1

### PR DESCRIPTION
**Pull request by candiduslynx**

### CODE REVIEW

**Review by valyankilyan — APPROVED** [review]

**Review by hermanschaaf — COMMENTED** [review]
Looks great, very nice and clean implementation 👍  I just left a couple of comments about things that might be good to address before release.

File: client/services.go [hunk 1]
```diff
@@ -14,26 +13,29 @@ import (
...
-	IAM                     *iam.IAM
-	K8S                     *k8s.Kubernetes
-	KMS                     *kms.KMS
-	OrganizationManager     *organizationmanager.OrganizationManager
-	OrganizationManagerSAML *saml.OrganizationManagerSAML
-	ResourceManager         *resourcemanager.ResourceManager
-	VPC                     *vpc.VPC
-}
+type (
+	Services struct {
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply client/services.go:21]: Just used to the block tyle

File: resources/services/containerregistry/scan_results_fetch.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"context"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/yandex-cloud/cq-provider-yandex/client"
+	"github.com/yandex-cloud/go-genproto/yandex/cloud/containerregistry/v1"
+	"golang.org/x/sync/errgroup"
+)
+
+func fetchScanResults(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- interface{}) error {
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply resources/services/containerregistry/scan_results_fetch.go:12]: thanks!

File: gen/main.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"log"
+
+	"github.com/yandex-cloud/cq-provider-yandex/gen/recipies"
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply gen/main.go:6]: done another pr

File: resources/services/containerregistry/scan_results_fetch.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (candiduslynx) [reply resources/services/containerregistry/scan_results_fetch.go:12]: discarding here as the pre resolver has to have all resources created, and we actually create the resources ONLY after the list results for image logic.

**Comment by candiduslynx** [comment]
close in favor of https://github.com/yandex-cloud/cq-provider-yandex/pull/22

**Pull request closed without merging**